### DISCUSSION TITLE
Pasting over existing text in text area shouldn't assume it's a markdown link

### DISCUSSION THREAD

**Discussion by nicsrpi** [community/community#31764] [Other Feature Feedback, Questions, & Ideas] [answered]
## Current behaviour:
1. copy a URL to clipboard
2. highlight text in text area on GitHub (for example, when writing the description of a PR)
3. paste
4. GH magically inserts the markdown for a hyperlink (`[text you highlighted](url from clipboard)`) instead of deleting the highlighted text and replacing it with contents of clipboard

## Expected behaviour:
When highlighted text is pasted over, it should be deleted and replaced with the text that has been pasted. 

## Why:
To keep a consistent UX with almost every other text input area ever.

**Comment by spenserblack** [comment]
This feature has also annoyed me since I'm so used to the behavior of
>almost every other text input area ever.

I can see how some users would like this, though, so maybe this could be set in the settings page?

**Comment by DenverCoder1** [comment]
It's an intended feature, so it's unlikely it will be reverted.

Note that you can use <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>V</kbd> to bypass this shortcut feature and make it always replace the text instead.

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#links

**Comment by spenserblack** [comment, marked as answer]
It's now an optional feature: https://github.blog/changelog/2022-09-27-the-new-github-issues-september-27th-update/#%F0%9F%96%8C--paste-urls-formatted-or-as-plain-text

Go to [settings and look in accessibility](https://github.com/settings/accessibility).
